scripts/test-examples.js

```diff
@@ -1,21 +1,29 @@
 const path = require('path');
 const glob = require('glob');
 const execSync = require('child_process').execSync;
+const fs = require('fs');
 
-const examples = glob.sync(
-  path.join(
-    __dirname,
-    '..',
-    'examples',
-    '!(default-theme|e-commerce|media|tourism|hooks|hooks-ssr)'
-  )
-);
+const examples = glob
+  .sync(path.join(__dirname, '..', 'examples', '!(react-native*)'))
+  .map((examplePath) => {
+    const pkg = JSON.parse(
+      fs.readFileSync(path.join(examplePath, 'package.json'))
+    );
 
-examples.forEach((example) => {
-  execSync(
-    `cd ${example} && echo testing $(basename $(pwd)) && yarn && yarn build && yarn test`,
-    {
-      stdio: 'inherit',
-    }
-  );
-});
+    return {
+      path: examplePath,
+      shouldRunTest: Boolean(pkg.scripts && pkg.scripts.test),
+    };
+  });
+
+examples
+  .filter((example) => example.shouldRunTest)
```

Comment:
do we want to print a warning for an example without tests?

Reply:
I don't think so, as we didn't prior to this PR. We want to track that the examples that contain test are valid.